test/unit/latch.c

```diff
@@ -0,0 +1,114 @@
+#include "memory.h"
+#include "fiber.h"
+#include "latch.h"
+#include "unit.h"
+
+const size_t num_fibers = 3;
+
+static int
+order_f(va_list ap)
+{
+	size_t fid = *va_arg(ap, size_t *);
+	size_t *check = va_arg(ap, size_t *);
+	struct latch *latch = va_arg(ap, struct latch *);
+	latch_lock(latch);
+
+	printf("fiber %zu acquired the latch\n", fid);
```

Comment:
Debug print, should be removed? It messes with the test output.

Reply:
Ok, removed.